### ISSUE TITLE
[Feature Request] Example with LangGraph

### ISSUE THREAD

**Issue by nitin302** [temporalio/samples-python#146]
### Is your feature request related to a problem? Please describe.

How can we use temporal with LangGraph workflows - https://github.com/langchain-ai/langgraph? Could you give a example?

### Describe the solution you'd like

LangGraph is typically used in agentic workflows and using temporal would give us the ability to scale execution.

### Additional context

None

**Comment by cretz** [comment]
Temporal does not really care what technology you use _inside_ the activities, but the scaling of workflows/activities is controlled by Temporal. https://github.com/temporalio/samples-python/tree/main/langchain sample shows using LangChain in activities and you can do the same for invoking a LangGraph workflow inside an activity. But Temporal workflows and LangGraph workflows are unrelated.

**Comment by nitin302** [comment]
Yes, I understand that, but an example would be helpful.

**Comment by cretz** [comment]
Can you use the LangChain sample, but call LangGraph instead of a LangChain in the activity?

**Comment by nitin302** [comment]
Yes, but i am looking for how we can make LangGraph nodes/edges become workflows/activities to make it easier to understand... It would help folks building AI agents use temporal for scaling.

**Comment by cretz** [comment]
I don't think LangGraph nodes/edges become workflows/activities. I think you interact with the LangGraph library via an activity like any other library, but LangGraph primitives don't integrate in a special way with Temporal ones from what I can see at a quick glance. If there are specific suggestions to how integration could/should occur, that may help.

**Comment by eric-burel** [comment]
@cretz I feel like LangGraph users are not necessarily aware of the notion of "durable workflow" yet would be interested by it when they hit scaling issues. LangGraph or even LangChain can help with running computations in parallel or sequence at a certain scale but at some point you might indeed want to convert the nodes and edges to activities to operate workflows at a bigger scale. Temporal feels like a natural solution at this step.

**Comment by cretz** [comment]
I admit I am not that familiar with LangGraph and how uniquely it can integrate with Temporal vs any other external utility. We will leave this open and our AI efforts may get around to making this sample or it is open for contribution.
